error with vue-tsc 2.0.29

### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.17.0
- Nuxt Version:     3.12.4
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  npm@9.6.6
- Builder:          -
- User Config:      devtools, typescript, vue, modules, i18n, quasar, site, nitro, routeRules, sitemap, runtimeConfig, css, security
- Runtime Modules:  @nuxtjs/i18n@8.3.1, @pinia/nuxt@0.5.1, @pinia-plugin-persistedstate/nuxt@1.2.0, @nuxtjs/seo@2.0.0-rc.10, nuxt-quasar-ui@2.1.2, nuxt-security@2.0.0-rc.9
- Build Modules:    -
------------------------------

### Reproduction

it looks like auto import is not working


### Describe the bug

if i upgrade vue-tsc version,

i can't npm run build or dev

because I encounter errors with composable... etc.




`components/recruit/Table.vue:29:11 - error TS2339: Property 'convertUnixtoYMD' does not exist on type 'CreateComponentPublicInstance<Readonly<ExtractPropTypes<__VLS_TypePropsToOption<{ searchStr: string; data: RecruitList[]; }>>>, {}, unknown, ... 16 more ..., {}>'.

29           convertUnixtoYMD(item.endTime)
             ~~~~~~~~~~~~~~~~
`
`components/solution/Intro.vue:40:21 - error TS2339: Property 'navigateTo' does not exist on type 'CreateComponentPublicInstance<Readonly<ExtractPropTypes<__VLS_TypePropsToOption<{ data: Res_Solution; }>>>, { t: ComposerTranslation<{ [x: string]: LocaleMessage<VueMessageType>; }, ... 4 more ..., string>; }, ... 17 more ..., {}>'.

40             @click="navigateTo({ path: '/customer/brochure' })"
                       ~~~~~~~~~~
`

it looks like auto import is not working


### Additional context

_No response_

### Logs

_No response_

--- a/packages/nuxt/src/app/types/augments.d.ts
+++ b/packages/nuxt/src/app/types/augments.d.ts
@@ -51,3 +51,23 @@ declare module 'vue' {
     head?(nuxtApp: NuxtApp): UseHeadInput
   }
 }
+
+declare module '@vue/runtime-core' {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface App<HostElement> {
+    $nuxt: NuxtApp
+  }
+  interface ComponentCustomProperties {
+    $nuxt: NuxtApp
+  }
+}
+
+declare module '@vue/runtime-dom' {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface App<HostElement> {
+    $nuxt: NuxtApp
+  }
+  interface ComponentCustomProperties {
+    $nuxt: NuxtApp
+  }
+}

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -130,6 +130,14 @@ declare module '#app' {
   }
 }
 
+declare module '@vue/runtime-core' {
+  interface ComponentCustomProperties extends NuxtAppInjections { }
+}
+
+declare module '@vue/runtime-dom' {
+  interface ComponentCustomProperties extends NuxtAppInjections { }
+}
+
 declare module 'vue' {
   interface ComponentCustomProperties extends NuxtAppInjections { }
 }

